File: LEETCODE/LEETCODE-JAVA/74.java
//74. Search a 2D Matrix
//https://leetcode.com/problems/search-a-2d-matrix/

//Method-1 (using Binary search)
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m=matrix.length;
        int i=0, j=matrix[0].length-1;

        while(i<m && j>=0){
            if(matrix[i][j]==target){
                return true;
            }
            else if(target<matrix[i][j]){
                j--;
            }
            else{
                i++;
            }
        }

        return false;
    }
}

File: LEETCODE/LEETCODE-PYTHON/74.py
# 74. Search a 2D Matrix
# https://leetcode.com/problems/search-a-2d-matrix/description/

# Method-1 using binary search (TC-O(mn))
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m,i,j= len(matrix), 0, len(matrix[0])-1

        while i<m and j>=0:
            if matrix[i][j]==target:
                return True
            elif matrix[i][j]<target:
                i+=1
            else:
                j-=1

        return False
            